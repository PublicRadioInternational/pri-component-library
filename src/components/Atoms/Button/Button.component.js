/**
 * @file Button.component.js
 * Exports a classic button component.
 */

import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import startCase from 'lodash/startCase';
import styles from './Button.css';

console.log('Styles: %o', styles);

const cx = classNames.bind(styles);

/**
 * Component that renders a button with a click handler.
 */
const Button = props => {
  const {
    children,
    className: classNameProp,
    color,
    component,
    disabled,
    small,
    type,
    ...other
  } = props;
  // Generate a class name based on the color.
  const className = cx(
    {
      [`btn${startCase(color)}`]: !small,
      [`btnMobile${startCase(color)}`]: small
    },
    classNameProp
  );

  const buttonProps = {};

  let ComponentProp = component;

  if (ComponentProp === 'button' && other.href) {
    ComponentProp = 'a';
  }

  if (ComponentProp === 'button' || ComponentProp === 'input') {
    buttonProps.type = type;
    buttonProps.disabled = disabled;
  } else {
    buttonProps.role = 'button';
  }

  return (
    <ComponentProp className={className} {...buttonProps} {...other}>
      {children}
    </ComponentProp>
  );
};

Button.colors = ['white', 'orange', 'dark', 'blue'];

Button.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  color: PropTypes.oneOf(Button.colors),
  /**
   * The component used for the root node.
   * Either a string to use a DOM element or a component.
   */
  component: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.func,
    PropTypes.object
  ]),
  disabled: PropTypes.bool,
  small: PropTypes.bool,
  type: PropTypes.string
};

Button.defaultProps = {
  children: null,
  className: null,
  color: Button.colors[0],
  component: 'button',
  disabled: false,
  small: false,
  type: 'button'
};

export default Button;
