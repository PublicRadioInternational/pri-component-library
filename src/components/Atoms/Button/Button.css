/* import colors... */
@value colors: "../../00_global/colors.css";
@value white, whiteDark, grayLighter, gray, grayLight, grayDark, grayDarker, orange, orangeDark, orangeDarker, blue, blueMed, blueDark, yellow from colors;
/* import fonts... */
@value fonts: "../../00_global/fonts.css";
@value copy from fonts;
/* import breakpoints */
@value header as bp-header from "../../00_global/breakpoints.css";

/* Generic */
@define-mixin btn {
  background-color: whiteDark;
  border: 1px solid grayLighter;
  border-radius: 0.25rem;
  color: grayDark;
  cursor: pointer;
  display: inline-block;
  font: 700 1rem copy;
  letter-spacing: 0.05rem;
  line-height: 2.25;
  margin: 0;
  padding: 0 0.85rem;
  text-align: center;
  text-decoration: none;
  text-transform: uppercase;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out,
    border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}

@define-mixin btnWhiteHover {
  background-color: white;
}

@define-mixin btnWhiteActive {
  background-color: grayDark;
  color: white;
}

/* Inherits button always */
.btn,
.btnWhite,
.btnOrange,
.btnDark,
.btnBlue {
  @mixin btn;
}

.btn:hover,
.btnWhite:hover {
  @mixin btnWhiteHover;
}

.btn:active,
.btn[aria-expanded='true'],
.btnWhite:active,
.btnWhite[aria-expanded='true'] {
  @mixin btnWhiteActive;
}

/* Colors */
@define-mixin btnOrange {
  background-color: orange;
  border-color: yellow;
  color: white;
}

@define-mixin btnOrangeHover {
  background-color: orangeDark;
  color: white;
}

@define-mixin btnBlue {
  background-color: blue;
  border: 1px solid blueMed;
  color: white;
}

@define-mixin btnBlueHover {
  background-color: blueMed;
  color: white;
}

.btnOrange {
  @mixin btnOrange;
}

.btnOrange:hover,
.btnOrange:active,
.btnOrange[aria-expanded='true'] {
  @mixin btnOrangeHover;
}

/* Colors */
@define-mixin btnDark {
  background-color: grayDarker;
  border-color: transparent;
  color: white;
}

@define-mixin btnDarkHover {
  background-color: orangeDarker;
  color: white;
}

.btnDark {
  @mixin btnDark;
}

.btnDark:hover,
.btnDark:active,
.btnDark[aria-expanded='true'] {
  @mixin btnDarkHover;
}

/* Colors */
@define-mixin btnBlue {
  background-color: blue;
  border-color: transparent;
  color: white;
}

@define-mixin btnBlueHover {
  background-color: orange;
  color: white;
}

.btnBlue {
  @mixin btnBlue;
}

.btnBlue:hover,
.btnBlue:active,
.btnBlue[aria-expanded='true'] {
  @mixin btnBlueHover;
}

/* Mobile Icon versions */
.btnMobileWhite,
.btnMobileDark {
  color: grayLight;
}

.btnMobileWhite:hover {
  color: grayDark;
}

.btnMobileOrange {
  color: orange;
}

.btnMobileBlue {
  color: blue;
}

@media bp-header {
  .btnMobileWhite,
  .btnMobileOrange,
  .btnMobileDark,
  .btnMobileBlue {
    @mixin btn;
  }

  .btnMobileWhite:hover {
    @mixin btnWhiteHover;
  }

  .btnMobileWhite:active,
  .btnMobileWhite[aria-expanded='true'] {
    @mixin btnWhiteActive;
  }

  .btnMobileOrange {
    @mixin btnOrange;
  }

  .btnMobileBlue {
    @mixin btnBlue;
  }

  .btnMobileOrange:hover,
  .btnMobileOrange:active,
  .btnMobileOrange[aria-expanded='true'] {
    @mixin btnOrangeHover;
  }

  .btnMobileDark {
    @mixin btnDark;
  }

  .btnMobileDark:hover,
  .btnMobileDark:active,
  .btnMobileDark[aria-expanded='true'] {
    @mixin btnDarkHover;
  }

  .btnMobileBlue {
    @mixin btnBlue;
  }

  .btnMobileBlue:hover,
  .btnMobileBlue:active,
  .btnMobileBlue[aria-expanded='true'] {
    @mixin btnBlueHover;
  }
}

/* Input Specific */
.inputWrap {
  position: relative;
}

svg.inputIcon {
  cursor: pointer;
  left: 10px;
  margin-top: -9px;
  position: absolute;
  top: 50%;
}

.inputBtn {
  @composes: btnWhite;
}

.inputWithIcon {
  padding: 0.375rem 0.85rem 0.375rem 2rem;
}

.inputWithIcon:active {
  background-color: grayLighter;
  color: grayDark;
}

/* Buttons in children */
@define-mixin btnSmall {
  font: 0.875rem 'Raleway', sans-serif;
  letter-spacing: 0;
  padding: 0.5rem 0.7rem;
  text-transform: none;
}

.btnWrap input[type='submit'] {
  @mixin btn;
  @mixin btnOrange;
  @mixin btnSmall;
}

.btnWrap input[type='submit']:hover {
  @mixin btnOrangeHover;
}
