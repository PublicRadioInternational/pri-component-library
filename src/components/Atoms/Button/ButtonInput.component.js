/**
 * @file ButtonInput.component.js
 * Exports an input button component.
 */

import React from 'react';
import PropTypes from 'prop-types';
import styles from './Button.css';
import Icon from '../Icon/Icon.component';

/**
 * Component that renders an input button with a click handler.
 */
const ButtonInput = props => {
  const { onClick, className, value, color, svg, small } = props;
  // Generate a class name based on the color.
  const buttonClass = `btn${color}`;
  const buttonMobileClass = `btnMobile${color}`;

  return (
    <div className={styles.inputWrap}>
      {svg ? <Icon svg={svg} inline aria-hidden className="inputIcon" /> : null}
      <input
        className={`${
          small ? styles[buttonMobileClass] : styles[buttonClass]
        } ${className && className} ${styles.inputBtn} ${svg &&
          styles.inputWithIcon}`}
        type="submit"
        onClick={onClick}
        value={value}
      />
    </div>
  );
};

ButtonInput.propTypes = {
  onClick: PropTypes.func,
  value: PropTypes.string,
  color: PropTypes.oneOf(['Orange', 'White']),
  className: PropTypes.string,
  svg: PropTypes.oneOfType([PropTypes.object, PropTypes.string]),
  small: PropTypes.bool
};

ButtonInput.defaultProps = {
  color: 'White',
  className: null,
  value: null,
  onClick: () => {},
  svg: null,
  small: false
};

export default ButtonInput;
