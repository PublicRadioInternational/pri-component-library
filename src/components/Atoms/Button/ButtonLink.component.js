/**
 * @file ButtonLink.component.js
 * Exports a button link component.
 */

import React from 'react';
import PropTypes from 'prop-types';
import styles from './Button.css';

/**
 * Component that renders a link button with a click handler.
 */
const ButtonLink = props => {
  const { url, onClick, className, children, color, small } = props;
  // Generate a class name based on the color.
  const buttonClass = `btn${color}`;
  const buttonMobileClass = `btnMobile${color}`;

  return (
    <a
      href={url}
      className={`${
        small ? styles[buttonMobileClass] : styles[buttonClass]
      } ${className && className}`}
      onClick={onClick}
    >
      <span
        className={`${styles.textLabel} ${small && styles.textLabelMobile}`}
      >
        {children}
      </span>
    </a>
  );
};

ButtonLink.propTypes = {
  url: PropTypes.string,
  onClick: PropTypes.func,
  children: PropTypes.node,
  color: PropTypes.oneOf(['Orange', 'White']),
  className: PropTypes.string,
  small: PropTypes.bool
};

ButtonLink.defaultProps = {
  url: null,
  color: 'White',
  className: null,
  children: null,
  onClick: () => {},
  small: false
};

export default ButtonLink;
