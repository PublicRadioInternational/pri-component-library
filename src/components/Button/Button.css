.button {
  background-color: cyan;
  border: none;
  cursor: pointer;
  font-size: 1.1rem;
  font-weight: bold;
  padding: 0.5em 1em;
  text-transform: uppercase;
}

.button:hover {
  background-color: #eee;
}
