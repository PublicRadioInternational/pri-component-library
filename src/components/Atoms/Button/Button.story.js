/**
 * @file atoms.story.js
 * Story configuration for Atoms.
 */

import React from 'react';
import classNames from 'classnames/bind';
import { withInfo } from '@storybook/addon-info';
import { checkA11y } from '@storybook/addon-a11y';
import {
  withKnobs,
  text,
  select,
  boolean,
  object
} from '@storybook/addon-knobs/react';
import styles from '@sambego/storybook-styles';
import startCase from 'lodash/startCase';
import { shallow } from 'enzyme';
import expect from 'expect';
import {
  storiesOf,
  action,
  specs,
  describe,
  it,
  beforeEach,
  snapshot
} from '../../../../.storybook/facade';
import Button from './Button.component';
import btnStyles from './Button.css';

const cx = classNames.bind(btnStyles);

const stories = storiesOf('Atoms/Buttons', module);

// Convert flat colors array to object keyed by color for title cased label.
// Example: ['color'] to { color: 'Color' }.
const buttonColors = Button.colors.reduce(
  (obj, color) => ({ ...obj, [color.toLowerCase()]: startCase(color) }),
  {}
);

/**
 * Add storybook definition for buttons.
 */
stories.addDecorator(checkA11y);
stories.addDecorator(withKnobs);
stories.addDecorator(
  styles({
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    minHeight: '100vh',
    backgroundColor: '#fff',
    padding: '0 1rem'
  })
);

stories.add(
  'Info & Knobs',
  withInfo({
    inline: true,
    text: `
    Basic button UI element.
  `
  })(() => (
    <Button
      onClick={action('onClickHandler')}
      color={select('Color', buttonColors, Button.colors[0].toLowerCase())}
      disabled={boolean('Disabled')}
      small={boolean('Small')}
      href={text('URL')}
      type={select('Input Type', ['button', 'submit', 'reset'])}
      {...object('Other Attributes', {}, 'BTNOPT-1')}
    >
      {text('Label', 'Listen')}
    </Button>
  ))
);

const exampleSpecs = storiesOf('Atoms/Buttons/Examples');

exampleSpecs.addDecorator(checkA11y);
exampleSpecs.addDecorator(
  styles({
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    minHeight: '100vh',
    backgroundColor: '#fff',
    padding: '0 1rem'
  })
);

exampleSpecs.add('Default Listen Button', () => {
  const story = <Button onClick={action('button-clicked')}>Listen</Button>;

  specs(() =>
    describe('Default Listen Button', () => {
      let output; // eslint-disable-line

      beforeEach(() => {
        output = shallow(story);
      });

      it('Should default to <button> element.', () => {
        expect(output.type()).toEqual('button');
      });

      it('Should have "Listen" label.', () => {
        expect(output.text()).toEqual('Listen');
      });

      it('Should have class "btnWhite".', () => {
        expect(output.hasClass(cx('btnWhite'))).toEqual(true);
      });

      it('Should have type "button".', () => {
        expect(output.prop('type')).toEqual('button');
      });
    })
  );

  snapshot('Default Listen Button', story);

  return story;
});

exampleSpecs.add('Orange Donate Link', () => {
  const story = (
    <Button onClick={action('button-clicked')} href="/donate" color="orange">
      Donate
    </Button>
  );

  specs(() =>
    describe('Orange Donate Button', () => {
      let output; // eslint-disable-line

      beforeEach(() => {
        output = shallow(story);
      });

      it('Should convert to <a> element.', () => {
        expect(output.type()).toEqual('a');
      });

      it('Should have "Donate" label.', () => {
        expect(output.text()).toEqual('Donate');
      });

      it('Should have class "btnOrange".', () => {
        expect(output.hasClass(cx('btnOrange'))).toEqual(true);
      });

      it('Should have href "/donate".', () => {
        expect(output.prop('href')).toEqual('/donate');
      });
    })
  );

  snapshot('Orange Donate Button', story);

  return story;
});

exampleSpecs.add('Blue Submit Button', () => {
  const story = (
    <Button onClick={action('button-clicked')} type="submit" color="blue">
      Submit
    </Button>
  );

  specs(() =>
    describe('Blue Submit Button', () => {
      let output; // eslint-disable-line

      beforeEach(() => {
        output = shallow(story);
      });

      it('Should default to <button> element.', () => {
        expect(output.type()).toEqual('button');
      });

      it('Should have "Submit" label.', () => {
        expect(output.text()).toEqual('Submit');
      });

      it('Should have class "btnBlue".', () => {
        expect(output.hasClass(cx('btnBlue'))).toEqual(true);
      });

      it('Should have type "submit".', () => {
        expect(output.prop('type')).toEqual('submit');
      });
    })
  );

  snapshot('Blue Submit Button', story);

  return story;
});

exampleSpecs.add('Dark Reset Input', () => {
  const story = (
    <Button
      onClick={action('button-clicked')}
      component="input"
      type="reset"
      color="dark"
      value="Reset"
    />
  );

  specs(() =>
    describe('Blue Submit Button', () => {
      let output; // eslint-disable-line

      beforeEach(() => {
        output = shallow(story);
      });

      it('Should convert to <input> element.', () => {
        expect(output.type()).toEqual('input');
      });

      it('Should have "Reset" label (as value.)', () => {
        expect(output.prop('value')).toEqual('Reset');
      });

      it('Should have class "btnDark".', () => {
        expect(output.hasClass(cx('btnDark'))).toEqual(true);
      });

      it('Should have type "reset".', () => {
        expect(output.prop('type')).toEqual('reset');
      });
    })
  );

  snapshot('Blue Submit Button', story);

  return story;
});
