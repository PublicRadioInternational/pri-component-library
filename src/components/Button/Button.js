import React, { Component } from 'react';
import './Button.css';

class Button extends Component {
  constructor(props) {
    super(props);
    this.state = { clicked: false };

    // This binding is necessary to make `this` work in the callback
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick() {
    this.setState(prevState => ({
      clicked: !prevState.clicked
    }));
  }

  render() {
    const isClicked = this.state.clicked;
    return (
      <button className="button" onClick={this.handleClick}>
        {isClicked ? 'Toggled' : 'Untoggled'}
      </button>
    );
  }
}

export default Button;
