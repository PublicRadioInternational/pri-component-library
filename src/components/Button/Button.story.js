/**
 * @file index.js
 * Story configuration for Button component.
 */

import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { checkA11y } from '@storybook/addon-a11y';
import Button from './Button';

storiesOf('Button', module)
  .addDecorator(checkA11y)
  .add('Toggle Button', () => (
    <Button onClick={action('button-clicked')}>Hello Button</Button>
  ));
