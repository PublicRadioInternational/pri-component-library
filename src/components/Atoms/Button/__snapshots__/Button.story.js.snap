// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blue Submit Button 1`] = `
<button
  className="btnBlue"
  disabled={false}
  onClick={undefined}
  type="submit"
>
  Submit
</button>
`;

exports[`Blue Submit Button 2`] = `
<button
  className="btnBlue"
  disabled={false}
  onClick={undefined}
  type="submit"
>
  Submit
</button>
`;

exports[`Blue Submit Button 3`] = `
<input
  className="btnDark"
  disabled={false}
  onClick={undefined}
  type="reset"
  value="Reset"
/>
`;

exports[`Dark Reset Input 1`] = `
<input
  className="btnDark"
  disabled={false}
  onClick={undefined}
  type="reset"
  value="Reset"
/>
`;

exports[`Default Listen Button 1`] = `
<button
  className="btnWhite"
  disabled={false}
  onClick={undefined}
  type="button"
>
  Listen
</button>
`;

exports[`Default Listen Button 2`] = `
<button
  className="btnWhite"
  disabled={false}
  onClick={undefined}
  type="button"
>
  Listen
</button>
`;

exports[`Info & Knobs 1`] = `
<div>
  <div
    style={Object {}}
  >
    <button
      className="btnWhite"
      disabled={false}
      href={undefined}
      onClick={undefined}
      type="button"
    >
      Listen
    </button>
  </div>
  <div
    style={undefined}
  >
    <div
      style={
        Object {
          "WebkitFontSmoothing": "antialiased",
          "backgroundColor": "#fff",
          "border": "1px solid #eee",
          "borderRadius": "2px",
          "boxShadow": "0px 2px 3px rgba(0, 0, 0, 0.05)",
          "color": "#444",
          "fontFamily": "-apple-system, \\".SFNSText-Regular\\", \\"San Francisco\\", BlinkMacSystemFont, \\"Segoe UI\\", \\"Roboto\\", \\"Oxygen\\", \\"Ubuntu\\", \\"Cantarell\\", \\"Fira Sans\\", \\"Droid Sans\\", \\"Helvetica Neue\\", \\"Lucida Grande\\", \\"Arial\\", sans-serif",
          "fontSize": "15px",
          "fontWeight": 300,
          "lineHeight": 1.45,
          "marginBottom": "20px",
          "marginTop": "20px",
          "padding": "20px 40px 40px",
        }
      }
    >
      <div
        style={
          Object {
            "marginBottom": 0,
          }
        }
      >
        <div
          style={
            Object {
              "WebkitFontSmoothing": "antialiased",
              "color": "#444",
              "fontFamily": "-apple-system, \\".SFNSText-Regular\\", \\"San Francisco\\", BlinkMacSystemFont, \\"Segoe UI\\", \\"Roboto\\", \\"Oxygen\\", \\"Ubuntu\\", \\"Cantarell\\", \\"Fira Sans\\", \\"Droid Sans\\", \\"Helvetica Neue\\", \\"Lucida Grande\\", \\"Arial\\", sans-serif",
              "fontSize": "15px",
            }
          }
        >
          Basic button UI element.
        </div>
      </div>
      <div>
        <h1
          style={
            Object {
              "borderBottom": "1px solid #EEE",
              "fontSize": "25px",
              "margin": "20px 0 0 0",
              "padding": "0 0 5px 0",
            }
          }
        >
          Story Source
        </h1>
        <pre
          className="css-glamorous-pre--10jsrye"
        >
          <div>
            <div>
              <div
                style={
                  Object {
                    "paddingLeft": 18,
                    "paddingRight": 3,
                  }
                }
              >
                <span
                  style={
                    Object {
                      "color": "#777",
                    }
                  }
                >
                  &lt;
                  Button
                </span>
                <span />
                <span
                  style={
                    Object {
                      "color": "#777",
                    }
                  }
                >
                  &gt;
                </span>
              </div>
              <div
                style={
                  Object {
                    "paddingLeft": 33,
                    "paddingRight": 3,
                  }
                }
              >
                <span
                  style={
                    Object {
                      "color": "#777",
                    }
                  }
                >
                  Listen
                </span>
              </div>
              <div
                style={
                  Object {
                    "paddingLeft": 18,
                    "paddingRight": 3,
                  }
                }
              >
                <span
                  style={
                    Object {
                      "color": "#777",
                    }
                  }
                >
                  &lt;/
                  Button
                  &gt;
                </span>
              </div>
            </div>
          </div>
          <button
            className="css-glamorous-button--jq496q"
            onClick={[Function]}
          >
            <div
              className="css-glamorous-div--ngxet1"
            >
              <div
                style={
                  Object {
                    "marginBottom": 6,
                  }
                }
              >
                Copied!
              </div>
              <div>
                Copy
              </div>
            </div>
          </button>
        </pre>
      </div>
      <div>
        <h1
          style={
            Object {
              "borderBottom": "1px solid #EEE",
              "fontSize": "25px",
              "margin": "20px 0 0 0",
              "padding": "0 0 5px 0",
            }
          }
        >
          Prop Types
        </h1>
        <div>
          <h2
            style={
              Object {
                "margin": "20px 0 0 0",
              }
            }
          >
            "
            Button
            " Component
          </h2>
          <table
            className="css-glamorous-table--3qhq7f"
          >
            <thead>
              <tr>
                <th
                  className="css-glamorous-th--1gvy3it"
                >
                  property
                </th>
                <th
                  className="css-glamorous-th--1gvy3it"
                >
                  propType
                </th>
                <th
                  className="css-glamorous-th--1gvy3it"
                >
                  required
                </th>
                <th
                  className="css-glamorous-th--1gvy3it"
                >
                  default
                </th>
                <th
                  className="css-glamorous-th--1gvy3it"
                >
                  description
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td
                  className="css-glamorous-td--58bwg8"
                >
                  children
                </td>
                <td
                  className="css-glamorous-td--58bwg8"
                >
                  <span />
                </td>
                <td
                  className="css-glamorous-td--42r361"
                >
                  -
                </td>
                <td
                  className="css-glamorous-td--42r361"
                >
                  <span
                    style={
                      Object {
                        "color": "#777",
                      }
                    }
                  >
                    null
                  </span>
                </td>
                <td
                  className="css-glamorous-td--42r361"
                />
              </tr>
              <tr>
                <td
                  className="css-glamorous-td--58bwg8"
                >
                  className
                </td>
                <td
                  className="css-glamorous-td--58bwg8"
                >
                  <span />
                </td>
                <td
                  className="css-glamorous-td--42r361"
                >
                  -
                </td>
                <td
                  className="css-glamorous-td--42r361"
                >
                  <span
                    style={
                      Object {
                        "color": "#777",
                      }
                    }
                  >
                    null
                  </span>
                </td>
                <td
                  className="css-glamorous-td--42r361"
                />
              </tr>
              <tr>
                <td
                  className="css-glamorous-td--58bwg8"
                >
                  color
                </td>
                <td
                  className="css-glamorous-td--58bwg8"
                >
                  <span />
                </td>
                <td
                  className="css-glamorous-td--42r361"
                >
                  -
                </td>
                <td
                  className="css-glamorous-td--42r361"
                >
                  <span
                    style={
                      Object {
                        "color": "#22a",
                        "wordBreak": "break-word",
                      }
                    }
                  >
                    white
                  </span>
                </td>
                <td
                  className="css-glamorous-td--42r361"
                />
              </tr>
              <tr>
                <td
                  className="css-glamorous-td--58bwg8"
                >
                  component
                </td>
                <td
                  className="css-glamorous-td--58bwg8"
                >
                  <span />
                </td>
                <td
                  className="css-glamorous-td--42r361"
                >
                  -
                </td>
                <td
                  className="css-glamorous-td--42r361"
                >
                  <span
                    style={
                      Object {
                        "color": "#22a",
                        "wordBreak": "break-word",
                      }
                    }
                  >
                    button
                  </span>
                </td>
                <td
                  className="css-glamorous-td--42r361"
                />
              </tr>
              <tr>
                <td
                  className="css-glamorous-td--58bwg8"
                >
                  disabled
                </td>
                <td
                  className="css-glamorous-td--58bwg8"
                >
                  <span />
                </td>
                <td
                  className="css-glamorous-td--42r361"
                >
                  -
                </td>
                <td
                  className="css-glamorous-td--42r361"
                >
                  <span
                    style={
                      Object {
                        "color": "#a11",
                      }
                    }
                  >
                    false
                  </span>
                </td>
                <td
                  className="css-glamorous-td--42r361"
                />
              </tr>
              <tr>
                <td
                  className="css-glamorous-td--58bwg8"
                >
                  small
                </td>
                <td
                  className="css-glamorous-td--58bwg8"
                >
                  <span />
                </td>
                <td
                  className="css-glamorous-td--42r361"
                >
                  -
                </td>
                <td
                  className="css-glamorous-td--42r361"
                >
                  <span
                    style={
                      Object {
                        "color": "#a11",
                      }
                    }
                  >
                    false
                  </span>
                </td>
                <td
                  className="css-glamorous-td--42r361"
                />
              </tr>
              <tr>
                <td
                  className="css-glamorous-td--58bwg8"
                >
                  type
                </td>
                <td
                  className="css-glamorous-td--58bwg8"
                >
                  <span />
                </td>
                <td
                  className="css-glamorous-td--42r361"
                >
                  -
                </td>
                <td
                  className="css-glamorous-td--42r361"
                >
                  <span
                    style={
                      Object {
                        "color": "#22a",
                        "wordBreak": "break-word",
                      }
                    }
                  >
                    button
                  </span>
                </td>
                <td
                  className="css-glamorous-td--42r361"
                />
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Orange Donate Button 1`] = `
<a
  className="btnOrange"
  href="/donate"
  onClick={undefined}
  role="button"
>
  Donate
</a>
`;

exports[`Orange Donate Link 1`] = `
<a
  className="btnOrange"
  href="/donate"
  onClick={undefined}
  role="button"
>
  Donate
</a>
`;
